bench/workgen/workgen.cxx

```diff
@@ -992,8 +993,21 @@ ThreadRunner::op_run(Operation *op)
         if (num_tables == 0)
             goto err;
 
-        tint = (random_value() % num_tables);
-        table_uri = _icontext->_dyn_table_names.at(tint);
+        // Select a random key.
+        auto it = _icontext->_dyn_tint.begin();
+        std::advance(it, random_value() % _icontext->_dyn_tint.size());
+        const std::string uri(it->first);
+
+        // Make sure the table is not marked for deletion.
+        bool marked_deleted =
+          std::find(_icontext->_dyn_tables_delete.begin(), _icontext->_dyn_tables_delete.end(),
```

Comment:
To avoid the complexity of the `std::find` function, should I transform the `_dyn_tables_delete` into an `std::map<std::string, bool>` and do:
```
marked_deleted = _dyn_tables_delete[uri];
```

Reply:
A std::map sounds like a good fit.

Reply:
I am not sure anymore what's the best... If I do:

```
marked_deleted = _dyn_tables_delete[uri];
```
Then if the `uri` key does not exist, it will get created... Which means another entry in the map will be added. I could do:
```
marked_deleted = _dyn_tables_delete.count(uri) && _dyn_tables_delete[uri];
```
The call to `count` would check if the key exists but it's basically a call to a `std::find`.
Finally, I could do:
```
marked_deleted = _dyn_tables_delete.at(uri);
```
But this means I need to handle the exception if the `uri` key does not exist.